### ISSUE TITLE
Changing the Battery Notes battery sensor name unexpectedly sets it to not-visible

### ISSUE THREAD

**Issue by chriswood1001** [andrew-codechimp/HA-Battery-Notes#921]
### System Health details

Version: 2024.2.0
Source: Container
Channel: Stable

I'm unable to find the "System Health" page on my Docker installation with those instructions. I've submitted a separate ticket to HA regarding this.

### Checklist

- [X] I have added `battery_notes:` to my configuration.yaml and restarted.
- [X] I have read the [FAQ's](https://andrew-codechimp.github.io/HA-Battery-Notes/faq).
- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of any [previous issues](https://github.com/andrew-codechimp/ha-battery-notes/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

Set the Battery Notes battery sensor to the same entity_id of the original devices battery sensor automatically hides both battery sensors when ```hide_battery: true```. My goal is to rename my old/original battery sensors (e.g append _original) and hide them, then make the Battery Notes sensor match the original sensor ID. This means existing automations or dashboards don't need to be edited while allowing Battery Notes to become my new reference.

### Reproduction steps

1. Enable ```hide_battery: true``` in your configuration.yaml and restart HA.
2. View a battery operated device and notice the devices original battery is hidden (expected) and Battery Notes battery sensor (with the various attributes) is displayed (expected).
3. Edit the entity_id of the **original device** battery sensor to something like: ```sensor.livingroom_temperature_battery_original```
4. Edit the entity_id of the **Battery Notes** sensor to match the original devices sensor id: ```sensor.livingroom_temperature_battery```
5. Observe that the Battery Notes sensor is now marked as (Hidden) with no ability to unhide it. The settings reads, "/!\ Hidden by Integration".

I've also tried performing steps 3 and 4 while ```hide_battery:false```, and then set ```hide_battery:true```; however, the issue persists on HA restart.
...

### Debug logs

```text
GitHub error: Issue templates and forms are currently unavailable. Please try again later.
```

**Comment by hellcry37** [comment]
Why would you rename my old/original battery sensors (e.g append _original) and then rename the battery plus one to _battery? I mean I understand that you dont want to edit your automations but isn't a bit crazy to do this?
I mean you work disable and rename original entities and to rename all the stuff in this integration to mach original names ... but you do not rename sensors in automations?

You can do this with VSCode in literary 2 minutes and a HA restart!

So basically you work twice as much, **just not to work on changing sensors in automations** ...
